feed/tests/test_workflowlevel1view.py

```diff
@@ -21,13 +22,133 @@ def test_create_workflowlevel1(self):
+        factory = APIRequestFactory()
+        data = {'name': 'Save the Lennons'}
+        request = factory.post('/api/workflowlevel1/', data)
+        request.user = self.tola_user.user
+        view = WorkflowLevel1ViewSet.as_view({'post': 'update'})
+        response = view(request, pk=wflvl1.pk)
+        self.assertEqual(response.status_code, 403)
+
+    def test_update_workflowlevel1_program_admin(self):
+        factory = APIRequestFactory()
+        data = {'name': 'Save the Children'}
+        request = factory.post('/api/workflowlevel1/', data)
+        request.user = self.tola_user.user
+        view = WorkflowLevel1ViewSet.as_view({'post': 'create'})
+        response = view(request)
+
+        factory = APIRequestFactory()
+        data = {'name': 'Save the Lennons'}
+        request = factory.post('/api/workflowlevel1/', data)
+        request.user = self.tola_user.user
+        view = WorkflowLevel1ViewSet.as_view({'post': 'update'})
+        response = view(request, pk=response.data['id'])
+        self.assertEqual(response.status_code, 200)
+
+        wflvl1 = WorkflowLevel1.objects.get(pk=response.data['id'])
+        self.assertEquals(wflvl1.name, data['name'])
+
+    def test_update_workflowlevel1_program_team(self):
+        wflvl1 = factories.WorkflowLevel1()
+        group_program_team = factories.Group(name=ROLE_PROGRAM_TEAM)
+        wflvl1.organization = self.tola_user.organization
+        wflvl1.user_access.add(self.tola_user)
+        wflvl1.save()
+        WorkflowTeam.objects.create(
+            workflow_user=self.tola_user, workflowlevel1=wflvl1,
+            role=group_program_team)
+
+        factory = APIRequestFactory()
+        data = {'name': 'Save the Lennons'}
+        request = factory.post('/api/workflowlevel1/', data)
+        request.user = self.tola_user.user
+        view = WorkflowLevel1ViewSet.as_view({'post': 'update'})
+        response = view(request, pk=wflvl1.pk)
+        self.assertEqual(response.status_code, 200)
+
+        wflvl1 = WorkflowLevel1.objects.get(pk=response.data['id'])
+        self.assertEquals(wflvl1.name, data['name'])
+
+    def test_update_workflowlevel1_same_org_different_program_team(self):
+        wflvl1_other = factories.WorkflowLevel1()
+        group_program_team = factories.Group(name=ROLE_PROGRAM_TEAM)
+        wflvl1_other.organization = self.tola_user.organization
+        wflvl1_other.user_access.add(self.tola_user)
+        wflvl1_other.save()
+        WorkflowTeam.objects.create(
+            workflow_user=self.tola_user, workflowlevel1=wflvl1_other,
+            role=group_program_team)
+
+        wflvl1 = factories.WorkflowLevel1(
+            organization=self.tola_user.organization)
+
+        factory = APIRequestFactory()
+        data = {'name': 'Save the Lennons'}
+        request = factory.post('/api/workflowlevel1/', data)
+        request.user = self.tola_user.user
+        view = WorkflowLevel1ViewSet.as_view({'post': 'update'})
```

Comment:
You use this API factory block several times with the same data, user and view action. Don't you think it's better to define it in the `setUp` and use it different test cases?